jumanji/environments/routing/pacman/env.py

```diff
@@ -0,0 +1,549 @@
+
+        # Check if episode terminates
+        num_pellets = next_state.pellets
+        dead = next_state.dead
+        time_limit_exceeded = next_state.step_count >= self.time_limit
+        all_pellets_found = num_pellets == 0
+        dead = next_state.dead == 1
+        done = time_limit_exceeded | dead | all_pellets_found
+
+        reward = jnp.asarray(collision_rewards)
+        action_mask_bool = jnp.array([True, True, True, True, False])
+        action_mask = self._compute_action_mask(next_state).astype(bool)
+        action_mask = action_mask * action_mask_bool
+
+        next_state = next_state.replace(action_mask=action_mask)  # type: ignore
+        # Generate observation from the state
+        observation = self._observation_from_state(next_state)
+
+        # Return either a MID or a LAST timestep depending on done.
+        timestep = jax.lax.cond(
+            done,
+            termination,
+            transition,
+            reward,
+            observation,
+        )
+
+        return next_state, timestep
+
+    def _update_state(self, state: State, action: chex.Array) -> Tuple[State, int]:
+        """Updates the state of the environment.
+
+        Args:
+            state: 'State` object corresponding to the new state of the environment.
+            action: An integer representing the player action.
+
+        Returns:
+            state: 'State` object corresponding to the new state of the environment.
+            collision_rewards: Rewards from objects the player has collided with
+        """
+
+        key = state.key
+        key, _ = jax.random.split(key)
+
+        # Move player
+        next_player_pos = player_step(state=state, action=action, steps=1)
+        next_player_pos = self.check_wall_collisions(state, next_player_pos)
+        state = state.replace(last_direction=jnp.array(action, jnp.int32))  # type: ignore
+
+        # Move ghosts
+        old_ghost_locations = state.ghost_locations
+        ghost_paths, ghost_actions, key = ghost_move(state)
+
+        # Check for collisions with ghosts
+        state, done, ghost_col_rewards = check_ghost_collisions(
+            ghost_paths, next_player_pos, state
+        )
+
+        state = state.replace(player_locations=next_player_pos)  # type: ignore
+        state = state.replace(dead=done)  # type: ignore
+
+        power_up_locations, eat, power_up_rewards = self.check_power_up(state)
+
+        # Check for collected pellets
+        collision_rewards, pellets, num_pellets = self.check_rewards(state)
+
+        # Update old ghost locations
+        state = state.replace(ghost_init_steps=state.ghost_init_steps - 1)  # type: ignore
+        state = state.replace(old_ghost_locations=old_ghost_locations)  # type: ignore
+
+        state = state.replace(pellet_locations=pellets)  # type: ignore
+        state = state.replace(pellets=num_pellets)  # type: ignore
+        state = state.replace(key=key)  # type: ignore
```

Comment:
To get rid of these type ignores just type all of these things as `chex.Array`. Unfortunately I don't there's a way around this with jax at the moment :(

Reply:
The linter seems to be fine with the ignores removed the first 2 state replaces will have it